src/pudl/transform/eia176.py

```diff
@@ -0,0 +1,116 @@
+"""Module to perform data cleaning functions on EIA176 data tables."""
+
+import pandas as pd
+from dagster import AssetCheckResult, AssetIn, AssetOut, asset_check, multi_asset
+
+from pudl.logging_helpers import get_logger
+
+logger = get_logger(__name__)
+
+
+@multi_asset(
+    outs={
+        "core_eia176__yearly_company_data": AssetOut(),
+        "core_eia861__yearly_aggregate_data": AssetOut(),
+    },
+)
+def _core_eia176__data(
+    raw_eia176__data: pd.DataFrame,
+) -> tuple[pd.DataFrame, pd.DataFrame]:
+    """Take raw list and return two wide tables with primary keys and one column per variable.
+
+    One table with data for each year and company, one with state- and US-level aggregates per year.
+    """
+    raw_eia176__data["report_year"].astype(int)
```

Comment:
These `astype` calls don't mutate the original object:

```
>>> import pandas as pd
>>> df = pd.DataFrame({"a": [1]})
>>> df.dtypes
a    int64
dtype: object
>>> df["a"].astype("float")
0    1.0
Name: a, dtype: float64
>>> df.dtypes
a    int64
dtype: object
```

So I think you'll have to do some reassignment.

I also think it would be nice to do all that reassignment in one call - which you can do by calling `astype` on the `DataFrame` instead of its constituent `Series`:

```python
>>> df = pd.DataFrame({"a": [1], "b": ["1"], "c": 1.0})
>>> df
   a  b    c
0  1  1  1.0
>>> df.dtypes
a      int64
b     object
c    float64
dtype: object
>>> df = df.astype({"a": "float", "b": "float"})
>>> df
     a    b    c
0  1.0  1.0  1.0
>>> df.dtypes
a    float64
b    float64
c    float64
dtype: object
```